api/v1beta1/hostedcluster_types.go

```diff
@@ -737,6 +737,76 @@ type KubevirtPlatformSpec struct {
 	// the same cluster and namespace as the Hosted Control Plane.
 	// +optional
 	Credentials *KubevirtPlatformCredentials `json:"credentials,omitempty"`
+
+	// StorageDriver defines how the KubeVirt CSI driver exposes StorageClasses on
+	// the infra cluster (hosting the VMs) to the guest cluster.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="storageDriver is immutable"
```

Comment:
I wonder if only this validation is necessary?  But I guess the others probably don't hurt